app/src/main/java/com/google/android/sambadocumentsprovider/browsing/broadcast/BroadcastBrowsingProvider.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.google.android.sambadocumentsprovider.browsing.broadcast;
+
+import android.util.Log;
+
+import com.google.android.sambadocumentsprovider.BuildConfig;
+import com.google.android.sambadocumentsprovider.browsing.BrowsingException;
+import com.google.android.sambadocumentsprovider.browsing.NetworkBrowsingProvider;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class BroadcastBrowsingProvider implements NetworkBrowsingProvider {
+  private static final String TAG = "BroadcastBrowsing";
+  private static final int LISTENING_TIMEOUT = 3000;
+
+  private final ExecutorService mExecutor = Executors.newFixedThreadPool(2);
+
+  private int mTransId = 0;
+
+  @Override
+  public List<String> getServers() throws BrowsingException {
+    try {
+      DatagramSocket socket = new DatagramSocket();
+
+      mTransId++;
+
+      mExecutor.submit(new BroadcastSender(socket, mTransId));
+
+      Future<List<String>> servers = mExecutor.submit(
+              new BroadcastListener(socket, mTransId, LISTENING_TIMEOUT));
+
+      return servers.get();
+    } catch(ExecutionException | InterruptedException | SocketException e){
```

Comment:
nit: leave a space between catch and parenthesis. Also leave a space between the parenthesis and the brace.